--- vendor/mruby-source/src/vm.c	2020-02-18 12:15:54.346444209 +0900
+++ vendor/mruby-source/src/vm.c	2020-02-27 15:10:14.387252963 +0900
@@ -117,7 +117,7 @@ stack_clear(mrb_value *from, size_t coun
 }
 
 static inline void
-stack_copy(mrb_value *dst, const mrb_value *src, size_t size)
+stack_copy(mrb_value *dst, const mrb_value *src, mrb_int size)
 {
   while (size-- > 0) {
     *dst++ = *src++;
